packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -6,6 +6,41 @@ import 'package:meta/meta.dart';
 
 import 'print.dart';
 
+/// The various priority levels used to filter which diagnostics are shown and
+/// omitted.
+///
+/// Trees of Flutter diagnostics can be very large so filtering the Diagnostics
+/// shown matters. Typically filtering to only show diagnostics with at least
+/// level debug is appropriate.
+enum DiagnosticLevel {
+  /// Use this level for diagnostic properties that match their default value
+  /// and other cases where showing a diagnostic would not add value.
+  fine,
+  /// Diagnostics that should only be shown when performing fine grained
+  /// debugging of an object.
+  ///
+  /// Used by properties that may sometimes be relevant but where the property
+  /// value is too verbose (e.g. 300 characters long) to show with a higher
+  /// diagnostic level.
+  debug,
+  /// Interesting diagnostics that should be typically shown.
+  info,
+  /// Very important diagnostics that indicate problematic property values.
+  ///
+  /// For example, use if you would write the property description
+  /// message in ALL CAPS.
```

Comment:
there are some examples of this in object.dart and box.dart, should they be updated too?

Reply:
updated the example in box.dart
the object.dart example doesn't quite apply as NEEDS-LAYOUT, NEEDS-PAINT and DETATCHED
aren't currently possible to handle as properties as they are in the toStringShort instead of the regular body of properties.We could be clever and display warning level FlagProperties first in the toString output if we wanted.